### PR TITLE
Compile code before publish

### PR DESCRIPTION
**Pull request by couto**
First of all, I'm sorry for such a big PR, this started with small bugfixes and then it became a coding session with trial and errors until I was able to use it on my project.

* Most people don't pass their node_module dependencies throught babel, so having code in React and using features that haven't yet landed on browsers is not really desirabled. 
In order to fix that, this PR adds a prepublish task that will compile the code into the dist folder. The compiled code no longer contains JSX nor class properties.

* We can use the `main` property from package.json to point to the entry file. We can take advantage of that to not have to point directly to the decorator.js file. So we can simply do `import { withThemes } from 'storybook-styled-components';`, also added the `jsnext:main` property so that bundlers like webpack can pick the uncompiled files

* Renamed the decorator function to `withThemes` this is more consistent with the most common addons (withKnobs, withNotes... )

* Split the Panel and the Wrap component into different files. This wasn't really necessary but it makes it more consistent when debugging several addons.

* Removed class properties. This leaves us with a more idiomatic code and 

Feel free to disregard this PR, I have touched in almost every file, so this is not really a clean PR, like I said... It started with small fixes and then this became a coding session.

### CODE REVIEW

**Comment by taddei** [comment]
Hi there, sorry for the late reply. I'll definitely have a look at the PR and merge if still compatible with our storybook 👍

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+      themes={themes}
+      channel={channel}
+      >{storyFn(context)}
+    </WrapStyledComponentsThemePicker>
+  );
+}
+
+export function withThemes(themes, defaultTheme) {
+  return (storyFn, context) => wrapperTheme(themes, defaultTheme)(storyFn, context)
+}
```
Comment (taddei) [inline src/index.js]: how about cleaning this up a little? like
```javascript
export const withThemes = themes => (storyFn, context) => {
  const channel = addons.getChannel();
  return (
    <WrapStyledComponentsThemePicker
      themes={themes}
      channel={channel}
      >{storyFn(context)}
    </WrapStyledComponentsThemePicker>
  )
}
```

File: src/WrapStory.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { ThemeProvider } from 'styled-components'
+
+export default class WrapStyledComponentsThemePicker extends React.Component {
```
Comment (taddei) [inline src/WrapStory.js]: how about we call it like the file name? `class WrapStory extends React.Component` ?

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import React from 'react'
+import addons from '@storybook/addons'
+import WrapStyledComponentsThemePicker from './WrapStory'
```
Comment (taddei) [inline src/index.js]: `WrapStory` as above ?

**Review by taddei — CHANGES_REQUESTED** [review]
Looking good. What do you think of the changes?

File: src/index.js (hunk 1 above)
Comment (couto) [reply src/index.js]: I don't have the project with me at this right moment, but I remember that I tried that approach before and somehow it failed. 
However like the PR states, a lot of things changed. 
I can give it another try and if it works, I'm pretty ok with the change. 👍

**Comment by couto** [comment]
I've just committed with the requested changes, it seems to work on my computer, so I really can't remember what was wrong before. :D

**Comment by kristian-puccio** [comment]
Do you mind publishing this change?
Thanks

**Pull request merged**